contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+            }
+        }
+
+        // emit an event so it is easier to find correct values off-chain
+        (
+            uint256 _wethPositionBalance,
+            uint256 _oethbPositionBalance
+        ) = getPositionPrincipal();
+        emit PrincipalPositionBeforeSwap(
+            _wethPositionBalance,
+            _oethbPositionBalance
+        );
+
+        // Swap it
+        swapRouter.exactInputSingle(
+            // sqrtPriceLimitX96 is just a rough sanity check that we are within 0 -> 1 tick
+            // a more fine check is performed in _checkForExpectedPoolPrice
+            // TBD(!): this needs further work if we want to generalize this approach
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: address(_tokenToSwap),
+                tokenOut: _swapWeth ? OETHb : WETH,
+                tickSpacing: tickSpacing, // set to 1
+                recipient: address(this),
+                deadline: block.timestamp,
+                amountIn: _amountToSwap,
+                amountOutMinimum: _minTokenReceived, // slippage check
+                sqrtPriceLimitX96: _swapWeth
+                    ? sqrtRatioX96Tick0
+                    : sqrtRatioX96Tick1
+            })
+        );
+
+        // emit an event so it is easier to find correct values off-chain
+        (_wethPositionBalance, _oethbPositionBalance) = getPositionPrincipal();
+        emit PrincipalPositionAfterSwap(
+            _wethPositionBalance,
+            _oethbPositionBalance
+        );
+    }
+
+    /**
+     * @dev Add liquidity into the pool in the pre-configured WETH to OETHb share ratios
+     * configured by the `poolWethShare` property. This function will respect liquidity
+     * ratios when there no liquidity yet in the pool. If liquidity is already present
+     * then it relies on the `_swapToDesiredPosition` function in a step before to already
+     * move the trading price to desired position (with some tolerance).
+     */
+    // rebalance already has re-entrency checks
+    // slither-disable-start reentrancy-no-eth
+    function _addLiquidity() internal {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        require(_wethBalance > 0, "Must add some WETH");
+
+        uint160 _currentPrice = getPoolX96Price();
+        // sanity check active trading price is positioned within our desired tick
+        if (_currentPrice <= sqrtRatioX96Tick0 ||
+                _currentPrice >= sqrtRatioX96Tick1) {
+            int24 currentTick = getCurrentTradingTick();
+            revert OutsideExpectedTickRange(
+                currentTick,
+                lowerTick,
+                upperTick
+            );
+        }
+
+        // in case oethb would be the 1st token we'd need to call estimateAmount0 here
+        uint256 _oethbRequired = helper.estimateAmount1(
```

Comment:
🟣 Again this can revert if the current price has too little WETH. However we are unlikely to target such an amount.

See https://github.com/OriginProtocol/origin-dollar/pull/2184/files#r1727716358

Reply:
Yup also addressed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/8e22f76d984d50c7cfd6c0ee04dd42b38a98f4e9